src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    /// <summary>
+    /// Acquired lease
+    /// </summary>
+    public class AcquiredLease 
+    {
+        /// <summary>
+        /// The resource key which the lease is attached to 
+        /// </summary>
+        public string ResourceKey { get; }
+        /// <summary>
+        /// Duration of the aquired lease
+        /// </summary>
+        public TimeSpan Duration { get; }
+        /// <summary>
+        /// Lease token, whcih will be null if acquiring or renewing the lease failed
+        /// </summary>
+        public string Token { get; }
+        /// <summary>
+        /// Start time for this lease, which is when the lease is acquired or renewed
```

Comment:
I wonder if we should note that this time is in the callers time, so implementers don't assign it with the server side time.